### PR TITLE
Add Foreach node

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java [hunk 1]
```diff
@@ -298,6 +300,21 @@ public PropertiesBuilder expression(ExpressionNode expressionNode) {
             return this;
         }
 
+        public PropertiesBuilder<T> collection(Node expressionNode) {
+            semanticModel.typeOf(expressionNode).ifPresent(propertyBuilder::type);
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java:304]: We do not need it here. just build the property with `.type(Property.Type.EXPRESSION)`

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java [hunk 2]
```diff
@@ -558,6 +575,19 @@ public PropertiesBuilder<T> defaultExpression(ExpressionAttributes.Info info) {
             return this;
         }
 
+        public PropertiesBuilder<T> defaultCollection() {
+            Property property = propertyBuilder
+                    .metadata()
+                    .label(Property.COLLECTION_LABEL)
+                    .description(Property.COLLECTION_DOC)
+                    .stepOut()
+                    .value("[]")
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java:584]:
```suggestion
                    .value("[]")
                    .type(Property.Type.EXPRESSION)
```

**Review by nipunayf — COMMENTED** [review]
Shall we add this node to the AvailableNodesGenerator as well?

**Comment by KavinduZoysa** [comment]
> Shall we add this node to the AvailableNodesGenerator as well?

Added a test

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java (hunk 1 above)
Comment (nipunayf) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/NodeBuilder.java:304]: @KavinduZoysa let's remove this line for now

**Review by nipunayf — APPROVED** [review]

**Pull request merged**